### ISSUE TITLE
Volume of historical data is always -1

### ISSUE THREAD

**Issue by bennycode** [stoqey/ib#114]
I am using `client.reqHistoricalData` but every candle that I get has a volume of `-1`. Am I doing something wrong here?

```ts
client
  .on(EventName.error, error => {
    reject(error);
  })
  .on(
    EventName.historicalData,
    (
      reqId: number,
      timeInSeconds: string,
      open: number,
      high: number,
      low: number,
      close: number,
      volume: number
    ) => {
      if (timeInSeconds.startsWith('finished')) {
        client.unsubscribeFromGroupEvents(reqId);
      } else {
        console.log('volume', volume);
      }
    }
  );

const useRegularTradingHours = 0;
const dateFormat = 2;

client.reqHistoricalData(
  1337,
  new Stock('AAPL', undefined, 'USD'),
  '20210810 17:00:00 UTC',
  `3600 S`,
  BarSizeSetting.HOURS_ONE,
  'BID_ASK',
  useRegularTradingHours,
  dateFormat,
  false
);
```

Tested with @stoqey/ib v1.2.5.

**Comment by mfrener** [comment]
Think it's  'BID_ASK' argument
That is the data you want to receive. If you lookup
https://interactivebrokers.github.io/tws-api/historical_bars.html#hd_what_to_show
you see that "BID_ASK" has N/A on Volume.
Try 'TRADES' instead
(hmm we should add an enum for those whatShowType and maybe add that table to docu).

btw feel free to join slack channel, usually faster way to get feedback ;)
https://github.com/stoqey/ib/discussions/26

**Comment by bennycode** [comment]
Thank you! I will join your Slack channel. 😊 Btw. how about flagging that in the API? If there is no volume I would expect volume to be optional in the listener: `volume?: number`

Alternatively you could work with function overloading to make sure that when people use `BID_ASK` no volume will be returned in the listener. The return of `-1` is a false friend.

**Comment by mfrener** [comment]
Thing is.. one of the design goals of IBApi is/was to stay as close as possible to the official IB TWS API, so porting to new version is less huzzle, we can always refer to IB doc or java source as you must do same in typescript. When we start to diverge, things get complicated... we did such a change with v1.2.0, where we made all MAX_VALUE undefined. We would need to do same for -1, but I really want to avoid because -1 is very common value and a simple if(v === 1) {v = undefined} is probably break a lot of things.
This is why we started IBApiNext (sits on top of IBApi). There we try to address those things and make that TWS API more user-friendly. 
Like you give IBApiNext.getHistoricalData a try. 
It returns a Promise<Bar[]>, where Bar is export interface Bar {
....
  /** The bar's traded volume if available (only available for TRADES). */
  volume?: number;
}
==> this is what you want, volume is either a valid number or undefined, nor more -1.

Here is an example App using it https://github.com/stoqey/ib/blob/master/src/tools/historical-data.ts
